### ISSUE TITLE
When nova uses diff guard

### ISSUE THREAD

**Issue by nenads** [kirschbaum-development/nova-mail#27]
> `/**
     * @return \Illuminate\Database\Eloquent\Relations\BelongsTo
     */
    public function sender()
    {
        return $this->belongsTo(config('auth.providers.users.model'));
    }`

Should sender be same model proider that is used by nova guard ?

**Comment by nenads** [comment]
something like this:
`        $guard = config('nova.guard');
        $senderModel = config('auth.guards.' . $guard . '.provider');
        return $this->belongsTo(config('auth.providers.'. $senderModel  .'.model'));`

**Comment by brandonferens** [comment]
@nenads thanks for your question. It seems like what you are wanting to do is use a different model for the sender than what is set as your base auth user model, correct?

**Comment by nenads** [comment]
Yes. I gess quick fix it to get Nova guard provider. Maybe ideal is to have
morph on sender to.

BR

**Comment by brandonferens** [comment]
We are working on a fix for this and may have it out next week, but as it is Christmas...
